twitter_api/main.py

```diff
@@ -24,5 +37,207 @@ def not_found(e):
 
 
 @app.errorhandler(401)
-def not_found(e):
+def unauth_request(e):
     return '', 401
+
+@app.route('/login', methods=['GET', 'POST']) 
+def login():
+    if request.method == 'POST':
+        # import ipdb;  ipdb.set_trace()
+        
+ 
+        data = json.loads(str(request.data.decode('utf-8')))
+      
+        # print md5( data['password']) 
+        if 'password' not in data:
+            return 'Why u no password?',400
+        cur = g.db.cursor()
+        cur.execute('select id from user where username = "%s";' % data['username'])
```

Comment:
Formatting the query this way with data provided by the user can cause a possible SQL Injection case. Instead, you can send kind of a "context" as a second param in the `cursor.execute()` method.

In this case it should be something like this:
```
query = """
    SELECT id
    FROM user
    WHERE username=:username;
"""
cur.execute(query, {'username': data['username']})
```